pkg/eks/api/api.go

```diff
@@ -54,50 +55,168 @@ type ClusterConfig struct {
+			CIDR: zoneCIDR,
+		}
+	}
 }
+
+// NewNodeGroup crears new nodegroup inside cluster config,
+// it returns pointer to the nodegroup for convenience
+func (c *ClusterConfig) NewNodeGroup() *NodeGroup {
+	ng := &NodeGroup{
+		ID:             len(c.NodeGroups),
+		SubnetTopology: SubnetTopologyPublic,
+	}
+
+	c.NodeGroups = append(c.NodeGroups, ng)
+
+	return ng
+}
+
+// NodeGroup holds all configuration attributes that are
+// specific to a nodegroup
+type NodeGroup struct {
+	ID int
+
+	AMI               string
+	InstanceType      string
+	AvailabilityZones []string
+	Tags              map[string]string
+	SubnetTopology    SubnetTopology
+
+	DesiredCapacity int
+	MinSize         int
+	MaxSize         int
+
+	VolumeSize int
+
+	MaxPodsPerNode int
+
+	PolicyARNs      []string
+	InstanceRoleARN string
+
+	AllowSSH         bool
+	SSHPublicKeyPath string
+	SSHPublicKey     []byte
+	SSHPublicKeyName string
+}
+
+type (
+	// ClusterVPC holds global subnet and all child public/private subnet
+	ClusterVPC struct {
+		Network              // global CIRD and VPC ID
+		SecurityGroup string // cluster SG
+		// subnets are either public or private for use with separate nodegroups
+		// these are keyed by AZ for conveninece
+		Subnets map[SubnetTopology]map[string]Network
+		// for additional CIRD associations, e.g. to use with separate CIDR for
+		// private subnets or any ad-hoc subnets
+		ExtraCIDRs []*net.IPNet
+	}
+	// SubnetTopology can be SubnetTopologyPrivate or SubnetTopologyPublic
+	SubnetTopology string
+	// Network holds ID and CIDR
+	Network struct {
+		ID   string
+		CIDR *net.IPNet
+	}
+)
+
+const (
+	// SubnetTopologyPrivate repesents privately-routed subnets
+	SubnetTopologyPrivate SubnetTopology = "Private"
+	// SubnetTopologyPublic repesents publicly-routed subnets
+	SubnetTopologyPublic SubnetTopology = "Public"
+)
+
+// SubnetIDs returns list of subnets
+func (c ClusterVPC) SubnetIDs(topology SubnetTopology) []string {
+	subnets := []string{}
+	for _, s := range c.Subnets[topology] {
+		subnets = append(subnets, s.ID)
+	}
+	return subnets
+}
+
+// SubnetTopologies returns list of topologies supported
+// by a given cluster config
+func (c ClusterVPC) SubnetTopologies() []SubnetTopology {
+	topologies := []SubnetTopology{}
+	for topology := range c.Subnets {
+		topologies = append(topologies, topology)
+	}
+	return topologies
+}
+
+// ImportSubnet loads a given subnet into cluster config
+func (c ClusterVPC) ImportSubnet(topology SubnetTopology, az, subnetID string) {
+	if _, ok := c.Subnets[topology]; !ok {
+		c.Subnets[topology] = map[string]Network{}
+	}
+	if network, ok := c.Subnets[topology][az]; !ok {
+		c.Subnets[topology][az] = Network{ID: subnetID}
+	} else {
+		network.ID = subnetID
+		c.Subnets[topology][az] = network
+	}
+}
+
+// HasSuffiecentPublicSubnets validates if there is a suffiecent
+// number of subnets available to create a cluster
+func (c ClusterVPC) HasSuffiecentPublicSubnets() bool {
```

Comment:
Nitpick here, but just noticed the typo: `Sufficient`

Reply:
Thanks - it's one of the words I always forget how to spell!